### ISSUE TITLE
Embed public keys in vCard / contact

### ISSUE THREAD

**Issue by h-2** [kontalk/androidclient#455]
I know there is a ticket for using OpenKeychain for key management, but I thought maybe it would be a lot smarter to just embed the public key in the vcard, according to wikipedia this is possible: https://en.wikipedia.org/wiki/VCard

It would have the great advantage that keys are backed up and distributed with regular contacts backup, they are preserved when kontalk is uninstalled a.s.o. This would mean that all of my applications, be they on the desktop or on different mobile devices would all share the keys and I don't have to repeatedly re-verify someone's key (I know that right now they can't be verified at all, but at some point we will need that).
Maybe other program could be convinced to use these, as well, like K-9 Mail. Kontalk could share the keypair with other PGP applications.

What do you think?

**Comment by daniele-athome** [comment]
If I understand correctly, you're talking about storing public keys in Android contacts, right? It should be what Openkeychain does already, although introducing one more extension by Kontalk would only cause more fragmentation.

**Comment by h-2** [comment]
No, OpenKeychain doesn't store inside the contacts, it stores seperately and only "links" the contacts in its own database with the others. And there is no sync with CardDAV (or anything else AFAICT).

**Comment by daniele-athome** [comment]
Oh...shame on them :-) I kind of took that for granted (I didn't check the code and I don't have contacts with keys at the moment). Anyway this is a request that should be addressed to Openkeychain don't you think?
HOWEVER - usually the Kontalk key is generated separately from any other personal key of other kind, so I will give it some thought and get back to you.

**Comment by h-2** [comment]
Ok, I will contact them about it. But I think they don't do it, because they really need a proper keyring with signatures and all of that (which would be too much to save in the vcard). But if you don't need to save signatures on keys, and especially if you use small ECC keys, I think saving them in the vcard would be pretty cool!

**Comment by daniele-athome** [comment]
Even if I store the public key into a DATA field in the contact database, there is no way to export it in the Android built-in vCard exporter, so I'll have to create something to generate the VCF manually so I can add the public key to it. There is [some effort to create a spec and an implementation for this](https://guardianproject.info/2013/12/28/integrating-crypto-identities-with-android/), but is far from being an Android standard yet.
@h-2 am I getting this right? Did you mean to have a PUBLICKEY field in the VCF of the contacts you're sharing right? If that's the case, is a medium-sized feature and will be delayed to 3.2 or even future developments :-(

**Comment by daniele-athome** [comment]
[Resurrection mode]

@h-2 pardon my insistence, did you read my last question? This issue is still pending and I'd like to triage it.
